api/middleware/authRequired.js

```diff
@@ -7,18 +7,22 @@ const oktaJwtVerifier = new OktaJwtVerifier(oktaVerifierConfig.config);
  * A simple middleware that asserts valid access tokens and sends 401 responses
  * if the token is not present or fails validation.
  */
-const authRequired = async (req, res, next) => {
-  try {
-    const authHeader = req.headers.authorization || '';
-    oktaJwtVerifier
-      .verifyAccessToken(authHeader, oktaVerifierConfig.expectedAudience)
-      .then(async () => {
-        next();
-      })
-      .catch(next);
-  } catch (err) {
-    next(createError(401, err.message));
+const authRequired = (req, res, next) => {
+  const authHeader = req.headers.authorization || '';
+  const match = authHeader.match(/Bearer (.+)/);
+  if (!match) {
+    throw new Error('No idToken');
   }
+  const idToken = match[1];
+  // eslint-disable-next-line prettier/prettier
+  oktaJwtVerifier
+    .verifyAccessToken(idToken, oktaVerifierConfig.expectedAudience)
+    .then(() => {
+      next();
+    })
+    .catch((err) => {
+      next(createError(401, err.message));
+    });
 };
```

Comment:
Great code Ahmed, this middleware is now working as intended calling and passing correct data. Not giving access to a user with an invalid or no token as this middle is supposed to do. Beautiful piece of code here